### ISSUE TITLE
surface.set_at() inverting color positions on a png surface

### ISSUE THREAD

**Issue by galexandreg** [pygame/pygame#1910]
### Problem
_Maybe this is "normal". I don't know much about SDL surface internals... But it seems odd that get_at() and set_at() don't behave the same._

**Setting individual pixel colors on a surface created from a png file will invert color positions. (BGR, not RGB)**
get_at() will return colors from the png as RGB, but when setting a color with set_at() and reading it back with get_at(), they will be inverted.
It works fine with a "normal" surface.

### Test
```
import pygame

pygame.init()

image = pygame.image.load("test_get_set.png")
surface = pygame.Surface((500, 500))
surface.fill((0, 0, 0))


print("With the png (red, green, blue):")
print(image.get_at((0, 0)))
print(image.get_at((0, 200)))
print(image.get_at((0, 400)))
print()
print("Set pixels to red, green, blue with set_at:")
image.set_at((0, 0), (255, 0, 0))
image.set_at((0, 200), (0, 255, 0))
image.set_at((0, 400), (0, 0, 255))
print(image.get_at((0, 0)))
print(image.get_at((0, 200)))
print(image.get_at((0, 400)))
print()
print("Normal surface :")
print("Set pixels to red, green, blue with set_at:")
surface.set_at((0, 0), (255, 0, 0))
surface.set_at((0, 200), (0, 255, 0))
surface.set_at((0, 400), (0, 0, 255))
print(surface.get_at((0, 0)))
print(surface.get_at((0, 200)))
print(surface.get_at((0, 400)))
```
The test png file:
![test_get_set](https://user-images.githubusercontent.com/45190261/83690799-6a5c6c80-a5bf-11ea-9118-d9aff9b4927f.png)

**Comment by illume** [comment]
Interesting. I wonder if this is related to the solarwolf png issue.

**Comment by MyreMylar** [comment]
Modified the test slightly so you can see the error visually and printed out some more info:

    import pygame
    
    pygame.init()
    
    window_surf = pygame.display.set_mode((500, 500))
    
    image = pygame.image.load("test_get_set.png")
    print("bitsize:", image.get_bitsize())
    print('shifts:', image.get_shifts())
    
    
    print("With the png (red, green, blue):")
    print(image.get_at((0, 0)))
    print(image.get_at((0, 200)))
    print(image.get_at((0, 400)))
    print()
    print("Set pixels to red, green, blue with set_at:")
    image.set_at((10, 10), pygame.Color(255, 0, 0))
    image.set_at((10, 200), pygame.Color(0, 255, 0))
    image.set_at((10, 400), pygame.Color(0, 0, 255))
    print("should be red  :", image.get_at((10, 10)))
    print("should be green:", image.get_at((10, 200)))
    print("should be blue :", image.get_at((10, 400)))
    
    window_surf.blit(image, (0, 0))
    
    running = True
    while running:
        for event in pygame.event.get():
            if event == pygame.QUIT:
                running = False
    
        pygame.display.update()


Output:

    bitsize: 24
    shifts: (0, 8, 16, 0)
    With the png (red, green, blue):
    (255, 0, 0, 255)
    (0, 255, 0, 255)
    (0, 0, 255, 255)

    Set pixels to red, green, blue with set_at:
    should be red  : (0, 0, 255, 255)
    should be green: (0, 255, 0, 255)
    should be blue : (255, 0, 0, 255)

And:

![image](https://user-images.githubusercontent.com/13382426/83805587-0cf01a80-a6a8-11ea-817c-65bf5821d414.png)


I had a look into surface.set_at() and it looks like it is handling endianness and bitshifting for 24 bit surfaces here:

https://github.com/pygame/pygame/blob/ead731fa0c4675333c4ec4a8b862e74ffe36e2c3/src_c/surface.c#L998-L1020

**Comment by MyreMylar** [comment]
Oooh is this because we are trying to shift things twice? It looks like https://wiki.libsdl.org/SDL_MapRGBA also does some accounting for the surface format so that may already have flipped the RGB order of the colour by the time we try and apply it to the surface pixels maybe?

**Comment by MyreMylar** [comment]
Had a look into it and this PR seems to fix the bug:

https://github.com/pygame/pygame/pull/1919

**Comment by MyreMylar** [comment]
I guess we could close this now, or just wait until the next release. Either way it seems to be fixed.

**Comment by nthykier** [comment]
Closing it as fixed by #1919 then. :)
